homeassistant/components/webostv/__init__.py

```diff
@@ -212,12 +208,64 @@ async def async_unload_entry(hass, entry):
     if unload_ok:
         hass.data[DOMAIN][DATA_CONFIG_ENTRY].pop(entry.entry_id)
         await hass_notify.async_reload(hass, DOMAIN)
-        client.clear_state_update_callbacks()
-        await client.disconnect()
+        await client.shutdown()
 
     # unregister service calls, check if this is the last entry to unload
     if unload_ok and not hass.data[DOMAIN][DATA_CONFIG_ENTRY]:
         for service in SERVICE_TO_METHOD:
             hass.services.async_remove(DOMAIN, service)
 
     return unload_ok
+
+
+class PluggableAction:
+    """A pluggable action handler."""
+
+    def __init__(self) -> None:
+        """Initialize."""
+        self._actions: dict[Any, AutomationActionType] = {}
+
+    def __bool__(self):
+        """Return if we have something attached."""
+        return bool(self._actions)
+
+    @callback
+    def async_attach(self, action: AutomationActionType, variables: dict[str, Any]):
+        """Attach a device trigger for turn on."""
+
+        @callback
+        def _remove():
+            del self._actions[_remove]
+
+        job = HassJob(action)
+
+        self._actions[_remove] = (job, variables)
+
+        return _remove
+
+    async def async_run(self, hass: HomeAssistant, context: Context | None = None):
```

Comment:
This doesn't need to be a coroutine function since we don't await inside.
```suggestion
    @callback
    def async_run(self, hass: HomeAssistant, context: Context | None = None):
```